react/week2/to-do-app/src/App.js

```diff
@@ -0,0 +1,35 @@
+import React, { useState } from 'react';
+import './App.css';
+import ToDoList from './ToDoList';
+
+const todoList = [
+  {id: 1,description: "Get out of bed"},
+  {id: 2,description: "Brush teeth"},
+  {id: 3,description: "Eat breakfast"},
+];
+let counter=todoList.length+1;
+
+function App() {
+  const onButtonClick = () => {
+    let newToDo={};
+    newToDo.id=counter++;
+    newToDo.description='Random ToDo';
+    setTodos(todos.concat(newToDo));
+  }
+
+  const handleDelete = (id) => {
+    const newTodos = [...todos];
+    newTodos.splice(id-1, 1);
+    setTodos(newTodos);
+  }
+
+  const [todos, setTodos] = useState(todoList);
+  return (
+    <div className="App">
+      <ToDoList list={todos} handleDelete={handleDelete}/>
```

Comment:
It's really nice that you have made the ToDoList into a component itself, that's a really good first step! In this case, however, I think you might as well just define the list, the handleDelete and the onButtonClick functionality inside the ToDoList component. As apps becomes larger, it's really important to think about "can this state and corresponding functions live inside my component or do I have to lift the state to the parent component".
In this case, App.js is the parent component and I would personally prefer to keep it as clean as possible. The todolist and the add/remove functionality does not depend on any other components and therefore it can live inside the ToDoList component itself. 

Sorry for the long comment - if anything is unclear, please just text me on Slack and I can elaborate 😄

Reply:
Thank you, i have resolved this part of code